### PR TITLE
Implemented search dialog for cities list in hotel activity

### LINKED ISSUE #149: Allow users to select City Name in hotels page
- Go to Navigation bar, click on `Travel`
- Select `Hotel Booking`
- Allow users to select a city

What to do?
- On the top, there is a bar showing `Showing Mumbai Hotels`.
- When user clicks on that bar, open a dialog ( popular with a list of popular cities from our server )
- Also add a search bar on that dialog that searches for the city starting with the typed name

Endpoints to use : 
- `get-all-cities`
- `get-city-by-name/city_prefix`

### PR DESCRIPTION
**Pull request by aayushinigam**
# Description
screenshot2
screenshot1

Fixes #149 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
Ran successfully on an android device.

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java [hunk 1]
```diff
@@ -206,16 +206,100 @@ public void onResponse(Call call, final Response response) throws IOException {
...
+                }
+            }
+        });
+        getHotelList();
+    }
+
     @Override
     protected void onResume() {
         super.onResume();
         mSource = mSharedPreferences.getString(SOURCE_CITY_ID, "1");
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java]: you can remove these 4 variables now. I don't think these are being used anywhere now:
```
private String mSource;
    private String mDestination;
    private String mSourceText;
    private String mDestinationText;
```

File: Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java [hunk 2]
```diff
@@ -55,6 +59,7 @@
 import static utils.Constants.MUMBAI_LON;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java]: We'll be able to remove these contants from Constants.java also then, 
```
public static final String SOURCE_CITY = "source_city";
    public static final String DESTINATION_CITY = "destination_city";

    public static final String SOURCE_CITY_ID = "source_city_id";
    public static final String DESTINATION_CITY_ID = "destination_city_id";
    public static final String DESTINATION_CITY_LAT = "destination_city_lat";
    public static final String DESTINATION_CITY_LON = "destination_city_lon";

    public static final String MUMBAI_LAT = "19.076";
    public static final String MUMBAI_LON = "72.8777";

```

File: Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java [hunk 3]
```diff
@@ -228,10 +312,23 @@ private boolean isCloseOnSingleTapDay() {
...
-                //TODO :: show a dialog with list of cities
+            case R.id.select_city:
+                new HotelSearchDialogCompat(HotelsActivity.this, getString(R.string.search_title),
+                        getString(R.string.search_hint), null, mSearchCities,
+                        new SearchResultListener<HotelSearchModel>() {
+                            @Override
+                            public void onSelected(BaseSearchDialogCompat dialog,
+                                                   HotelSearchModel item, int position) {
+                                String selectedCity = item.getId();
+                                String title = getString(R.string.showing) + " " + item.getName()
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java]: You can make use of placeholder string for making this : https://stackoverflow.com/questions/3656371/dynamic-string-using-string-xml

**Comment by Swati4star** [comment]
@aayushinigam I have added a few comments. Incorporate those.

and the UI looks great! 💯

**Comment by aayushinigam** [comment]
Sure! I'll make the changes. Thanks to the amazing library :)

**Pull request merged**